File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DockingManager/Functionality/PinWindowsPage.vb
Namespace CompanionKit.Controls.DockingManager.Functionality

	Public Class PinWindowsPage

        'User index global variable
        Private mintUserIndex As Integer = 0

        Public Sub New()
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles the Click event of the mobjPinButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjPinButton_Click(sender As Object, e As EventArgs) Handles mobjPinButton.Click
            'Pins all windows
            mobjDockingManager.PinAll()
        End Sub

        ''' <summary>
        ''' Handles the Click event of the objUnpinButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjUnpinButton_Click(sender As Object, e As EventArgs) Handles mobjUnpinButton.Click
            'Unpins all windows
            mobjDockingManager.UnpinAll()
        End Sub

        ''' <summary>
        ''' Handles the Load event of the PinWindowsPage control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub PinWindowsPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            'Fill up table adapter with data from table
            Me.mobjCustomersTableAdapter.Fill(Me.mobjNorthwindDataSet.Customers)

            'Creates 5 DockingWindows
            For i As Integer = 0 To 4
                Dim mobjDockingWindow As New DockingWindow()
                FillDockingWindow(mobjDockingWindow)
                mobjDockingManager.AddAutoHiddenWindows(DockStyle.Top, mobjDockingWindow)
            Next
        End Sub

        ''' <summary>
        ''' Fills the docking window.
        ''' </summary>
        ''' <param name="objDockingWindow">The obj docking window.</param>
        Private Sub FillDockingWindow(objDockingWindow As DockingWindow)
            'Fills string with User info
            Dim strUserInfo As String = String.Format("User Name:{0}" & vbCr & vbLf & "Company Name:{1}" & vbCr & vbLf & "Country:{2}" & vbCr & vbLf & "City:{3}" & vbCr & vbLf & "Address:{4}" & vbCr & vbLf & "Phone:{5}", mobjNorthwindDataSet.Customers(mintUserIndex).ContactName, mobjNorthwindDataSet.Customers(mintUserIndex).CompanyName, mobjNorthwindDataSet.Customers(mintUserIndex).Country, mobjNorthwindDataSet.Customers(mintUserIndex).City, mobjNorthwindDataSet.Customers(mintUserIndex).Address, _
             mobjNorthwindDataSet.Customers(mintUserIndex).Phone)
            'Creates and fills label control
            Dim mobjUserLabel As New Label()
            mobjUserLabel.AutoSize = False
            mobjUserLabel.Text = strUserInfo
            mobjUserLabel.Dock = DockStyle.Fill
            'Settings up DockingWindow control
            objDockingWindow.Text = mobjNorthwindDataSet.Customers(mintUserIndex).ContactName
            objDockingWindow.HeaderText = mobjNorthwindDataSet.Customers(mintUserIndex).ContactTitle
            objDockingWindow.Controls.Add(mobjUserLabel)
            'Increase counter or reset it if value more than customer count
            If (mintUserIndex = (mobjNorthwindDataSet.Customers.Count - 1)) Then
                mintUserIndex = 0
            Else
                mintUserIndex = mintUserIndex + 1
            End If
        End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DockingManager/Functionality/PinWindowsPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.DockingManager.Functionality
{
    public partial class PinWindowsPage : UserControl
    {
        //User index global variable
        int mintUserIndex = 0; 

        public PinWindowsPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Click event of the objPinButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjPinButton_Click(object sender, EventArgs e)
        {
            //Pins all windows
            mobjDockingManager.PinAll();
        }

        /// <summary>
        /// Handles the Click event of the objUnpinButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjUnpinButton_Click(object sender, EventArgs e)
        {
            //Unpins all windows
            mobjDockingManager.UnpinAll();
        }

        /// <summary>
        /// Handles the Load event of the PinWindowsPage control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void PinWindowsPage_Load(object sender, EventArgs e)
        {
            //Fill up table adapter with data from table
            this.mobjCustomersTableAdapter.Fill(this.mobjNorthwindDataSet.Customers); 

            //Creates 5 DockingWindows
            for (int i = 0; i < 5; i++)
            {
                DockingWindow mobjDockingWindow = new DockingWindow();
                FillDockingWindow(mobjDockingWindow);
                mobjDockingManager.AddAutoHiddenWindows(DockStyle.Top, mobjDockingWindow);
            }
        }

        /// <summary>
        /// Fills the docking window.
        /// </summary>
        /// <param name="objDockingWindow">The obj docking window.</param>
        void FillDockingWindow(DockingWindow objDockingWindow)
        {
            //Fills string with User info
            string strUserInfo = string.Format("User Name:{0}\r\nCompany Name:{1}\r\nCountry:{2}\r\nCity:{3}\r\nAddress:{4}\r\nPhone:{5}",
                mobjNorthwindDataSet.Customers[mintUserIndex].ContactName,
                mobjNorthwindDataSet.Customers[mintUserIndex].CompanyName,
                mobjNorthwindDataSet.Customers[mintUserIndex].Country,
                mobjNorthwindDataSet.Customers[mintUserIndex].City,
                mobjNorthwindDataSet.Customers[mintUserIndex].Address,
                mobjNorthwindDataSet.Customers[mintUserIndex].Phone);
            //Creates and fills label control
            Label mobjUserLabel = new Label();
            mobjUserLabel.AutoSize = false;
            mobjUserLabel.Text = strUserInfo;
            mobjUserLabel.Dock = DockStyle.Fill;
            //Settings up DockingWindow control
            objDockingWindow.Text = mobjNorthwindDataSet.Customers[mintUserIndex].ContactName;
            objDockingWindow.HeaderText = mobjNorthwindDataSet.Customers[mintUserIndex].ContactTitle;
            objDockingWindow.Controls.Add(mobjUserLabel);
            //Increase counter or reset it if value more than customer count
            mintUserIndex = mintUserIndex == (mobjNorthwindDataSet.Customers.Count - 1) ? 0 : mintUserIndex + 1;
        }
    }
}